### PR TITLE
Fix usgov support for communication services

### LINKED ISSUE #23719: azurerm_communication_service doesn't support US Gov
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.2

### AzureRM Provider Version

3.78.0

### Affected Resource(s)/Data Source(s)

azurerm_communication_service

### Terraform Configuration Files

```hcl
resource "azurerm_communication_service" "email" {
  name                = "email-${var.environment}"
  resource_group_name = "common"
  data_location       = "US Gov"
}
```

### Debug Output/Panic Output

```shell
Error: expected data_location to be one of ["Africa" "Asia Pacific" "Australia" "Brazil" "Canada" "Europe" "France" "Germany" "India" "Japan" "Korea" "Norway" "Switzerland" "UAE" "UK" "United States"], got US Gov
│ 
│   with azurerm_communication_service.email,
│   on email-manager.tf line 46, in resource "azurerm_communication_service" "email":
│   46:   data_location       = "US Gov"
```

### Expected Behaviour

I'd expect that we can pass in US Gov as a value to data_location.

### Actual Behaviour

I'm getting a validation failure on data_location.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by MichaelChristopherson**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR is resolving this issue here: https://github.com/hashicorp/terraform-provider-azurerm/issues/23719.  The `usgov` region now supports these resources and is currently only being stopped by the text reference validation.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)
- [x] (In progress) - Tested if I can create these resources manually 

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_communication_service` - support for `usgov` region added
* `azurerm_email_communication_service` - support for `usgov` region added

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #23719

### CODE REVIEW

**Comment by MichaelChristopherson** [comment]
Tested manually deploying a `azurerm_communication_service` into my usgov instance and was successfully able to do so.

**Review by katbyte — APPROVED** [review]
Thanks @MichaelChristopherson - LGTM 📆

File: internal/services/communication/email_service_resource.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ func (EmailCommunicationServiceResource) Arguments() map[string]*pluginsdk.Schem
 				"UAE",
 				"UK",
 				"United States",
+				"usgov"
```
Comment (katbyte) [inline internal/services/communication/email_service_resource.go:65]:
```suggestion
				"usgov",
```

File: internal/services/communication/communication_service_resource.go [hunk 2]
```diff
@@ -80,6 +80,7 @@ func (CommunicationServiceResource) Arguments() map[string]*pluginsdk.Schema {
 				"UAE",
 				"UK",
 				"United States",
+				"usgov"
```
Comment (katbyte) [inline internal/services/communication/communication_service_resource.go:83]:
```suggestion
				"usgov",
```

**Pull request merged**